app/src/main/java/com/github/se/polyfit/ui/navigation/NavigationGraph.kt

```diff
@@ -7,13 +7,19 @@ import androidx.navigation.compose.composable
 import androidx.navigation.navigation
 import com.github.se.polyfit.ui.components.GenericScreen
 import com.github.se.polyfit.ui.screen.OverviewScreen
+import com.github.se.polyfit.viewmodel.meal.MealViewModel
 
-fun NavGraphBuilder.globalNavigation(navController: NavHostController) {
+fun NavGraphBuilder.globalNavigation(
+    navController: NavHostController,
+    mealViewModel: MealViewModel
+) {
   navigation(startDestination = Route.Home, route = Route.Overview) {
     composable(Route.Home) {
       GenericScreen(
           navController = navController,
-          content = { paddingValues -> OverviewScreen(paddingValues) })
+          content = { paddingValues ->
+            OverviewScreen(paddingValues, navController, mealViewModel)
```

Comment:
instead of passing the navcontroller as argument pass a function of type ()->Unit with navController.navigate(Route.addMeal)

Reply:
The overview screen needs to navigate between different screens at the push of a button. Since there are multiple screens you can navigate to, a single function with no arguments cannot perform this task.